control-plane/stor-port/src/types/v0/transport/nexus.rs

```diff
@@ -79,6 +79,101 @@ impl From<Nexus> for models::Nexus {
 
 rpc_impl_string_uuid!(NexusId, "UUID of a nexus");
 
+/// A gRPC request type for creating snapshot of a nexus, which essentially
+/// means a snapshot of all(or selected) replicas associated with that nexus.
+pub struct CreateNexusSnapshot {
+    /// uuid of the nexus.
+    pub nexus: NexusId,
+    /// uuid of the entity, nexus in this case.
+    pub entity: uuid::Uuid,
+    /// A transaction id for this request.
+    pub txn_id: String,
+    /// name of the snapshot to be created.
+    pub name: String,
+    /// replica list descriptor.
+    pub replica_desc: Vec<CreateNexusSnapReplDescr>,
```

Comment:
nit: do all these fields need to be pub? You have getters below so they can stay private?